File: WOT-iOS/WOTData/WOTData/core/requests/Turrets/WOTWebRequestTankTurrets.m
//
//  WOTWebRequestTankTurrets.m
//  WOT-iOS
//
//  Created on 6/23/15.
//  Copyright (c) 2015. All rights reserved.
//

#import "WOTWebRequestTankTurrets.h"
#import "WOTDataDefines.h"
#import <WOTData/WOTData.h>

@implementation WOTWebRequestTankTurrets

- (NSDictionary *)query {
    
    NSMutableDictionary *result = [[NSMutableDictionary alloc] init];
    result[WOTApiKeys.application_id] = [NSString valueOrSpaceString:self.hostConfiguration.applicationID];
    result[WOTApiKeys.fields] = [NSString valueOrSpaceString:[self.args escapedValueForKey:WOTApiKeys.fields]];
    result[WOTApiKeys.module_id] = [NSString valueOrSpaceString:[self.args escapedValueForKey:WOTApiKeys.module_id]];
    return result;
}

- (NSString *)path {
    
    return @"/wot/encyclopedia/tankturrets/";
}

@end

File: WOT-iOS/WOTData/WOTData/WebServices/requests/WOTWEBRequestTankTurrets.swift
//
//  WOTWebRequestTankTurrets.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 1/14/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import Foundation

@objc
public class WOTWEBRequestTankTurrets: WOTWEBRequest {
    
    override public var method: String { return "POST" }

    override public var path: String {
        return "/wot/encyclopedia/tankturrets/"
    }
}

extension WOTWEBRequestTankTurrets: WOTModelServiceProtocol {
    
    @objc
    public class func modelClass() -> AnyClass? {
        return Tankturrets.self
    }
    
    @objc
    public func instanceModelClass() -> AnyClass? {
        return type(of: self).modelClass()
    }
}
